src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheckTest.java

```diff
@@ -807,4 +807,53 @@ public void testImportGroupsRedundantSeparatedInternally() throws Exception {
                 expected);
     }
 
+    @Test
+    public void testStaticGroupsBottom() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(ImportOrderCheck.class);
+        checkConfig.addAttribute("groups", "org, com, java, javax");
+        checkConfig.addAttribute("staticGroups", "org, com, java, javax");
+        checkConfig.addAttribute("option", "bottom");
+        checkConfig.addAttribute("ordered", "true");
+        checkConfig.addAttribute("separated", "true");
+        checkConfig.addAttribute("sortStaticImportsAlphabetically", "true");
+        checkConfig.addAttribute("useContainerOrderingForStatic", "false");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+
+        verify(checkConfig, getNonCompilablePath("InputImportOrderStaticGroupsBottom.java"),
+                expected);
+    }
+
+    @Test
+    public void testStaticGroupsTop() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(ImportOrderCheck.class);
+        checkConfig.addAttribute("groups", "org, com, java, javax");
+        checkConfig.addAttribute("staticGroups", "org, com, java, javax");
+        checkConfig.addAttribute("option", "top");
+        checkConfig.addAttribute("ordered", "true");
+        checkConfig.addAttribute("separated", "true");
+        checkConfig.addAttribute("sortStaticImportsAlphabetically", "true");
+        checkConfig.addAttribute("useContainerOrderingForStatic", "false");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+
+        verify(checkConfig, getNonCompilablePath("InputImportOrderStaticGroupsTop.java"),
+                expected);
+    }
+
+    @Test
+    public void testStaticGroupsNegative() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(ImportOrderCheck.class);
+        checkConfig.addAttribute("staticGroups", "");
+        checkConfig.addAttribute("option", "top");
+        checkConfig.addAttribute("ordered", "true");
+        checkConfig.addAttribute("separated", "true");
+        checkConfig.addAttribute("sortStaticImportsAlphabetically", "true");
+        checkConfig.addAttribute("useContainerOrderingForStatic", "false");
+        final String[] expected = {
+            "15: " + getCheckMessage(MSG_ORDERING, "java.lang.String.format"),
+        };
+
+        verify(checkConfig, getNonCompilablePath("InputImportOrderStaticGroupsNegative.java"),
+                expected);
```

Comment:
please add UT for:
1)
groups={<some groups>}
groupsStatic={<some groups>}
option= not top|bottom

There should be no effect of groupsStatic at all, no matter what value it has.

2)
groups={org, java}
groupsStatic={java, org}
option = top|bottom

to make sure that groupsStatic make its own effect .
and in input file put some imports that will fall to "other" group and should be just sorted or not (if sorting is enforced).

3)
groups={org, com, java, }
groupsStatic={}
option = top|bottom
sortStaticImportsAlphabetically=false

to make sure that groupsStatic make its own effect  to order "com, java, org" (alphabetically).

Reply:
Done: https://pbludov.github.io/issue-5724/index.html

Reply:
hmm, I requested UT update in code, not a report update.

Reply:
done.